### PR TITLE
Adapt to AppCompat v21

### LINKED ISSUE #586: Adapt app to AppCompat v21
- make BaseActivity inherit from v7.App.ActionBarActivity
- Use Theme.AppCompat as base theme

### PR DESCRIPTION
**Pull request by taavirehemagi**
- BaseActivity inherits from ActionBarActivity
- App theme to based on AppCompat instead of Holo
- Initial adding of Toolbar (needed for layouting purposes on Drawer redesign)
Closes #586

### CODE REVIEW

File: Joey/UI/Activities/MainDrawerActivity.cs [hunk 1]
```diff
@@ -221,36 +224,37 @@ private void SwitchActionBarView (int pageId)
 
         private void OpenPage (int id)
         {
-            SwitchActionBarView (id);
+
+//            SwitchActionBarView (id);
```
Comment (monday8am) [inline Joey/UI/Activities/MainDrawerActivity.cs]: The logic related with titles is commented and titles are not showed.

File: Joey/UI/Activities/MainDrawerActivity.cs [hunk 2]
```diff
@@ -90,10 +93,9 @@ protected override void OnCreateActivity (Bundle bundle)
 
             syncStatusText = DrawerSyncView.FindViewById<TextView> (Resource.Id.SyncStatusText);
 
-            ActionBar.SetCustomView (Timer.Root, lp);
-            ActionBar.SetDisplayShowCustomEnabled (true);
-            ActionBar.SetDisplayHomeAsUpEnabled (true);
-            ActionBar.SetHomeButtonEnabled (true);
+            MainToolbar = FindViewById<Toolbar> (Resource.Id.MainToolbar);
+            SetSupportActionBar (MainToolbar);
```
Comment (masterrr) [inline Joey/UI/Activities/MainDrawerActivity.cs]: I wonder how it works, Theme.Toggl.App <- Theme.Toggl.Base <- Theme.AppCompat.Light is used here.

If I call SetSupportActionBar when Resource.Style.Theme_AppCompat_Light_NoActionBar is used all is ok.
If I call SetSupportActionBar when Resource.Style.Theme_AppCompat_Light is used I get an exception:
“This Activity already has an action bar supplied by the window decor. Do not request Window.FEATURE_ACTION_BAR and set windowActionBar to false in your theme to use a Toolbar instead.”
https://github.com/toggl/mobile/pull/588/files#diff-24890dc1131a3a3d664e88682c652a93R21
https://github.com/toggl/mobile/pull/588/files#diff-0d0060d25b0d39ab16f29fd091a62be6R53


I don't see windowActionBar = false in our themes. 

Just some thinking, maybe this messes things up in the titles issue below.

**Comment by monday8am** [comment]
@masterrr 
> I don't see windowActionBar = false in our themes.
https://github.com/toggl/mobile/blob/taavirehemagi-redesign-appcompat/Joey/Resources/values/Styles.xml#L7

you refer to previous themes? Anyway, the key of this PR is that titles are missing. We need to recovery  them to close it correctly.

**Comment by taavirehemagi** [comment]
Got titles to work, what do you think?

**Comment by masterrr** [comment]
@taavirehemagi nice, I did the same way in the new project activity

**Comment by monday8am** [comment]
Rebased and ready to merge.

**Pull request merged**